client/pd_service_discovery.go

```diff
@@ -140,20 +148,32 @@ var (
 )
 
 type pdServiceClient struct {
-	addr       string
-	conn       *grpc.ClientConn
-	isLeader   bool
-	leaderAddr string
+	addr        string
+	httpAddress string
+	conn        *grpc.ClientConn
+	isLeader    bool
+	leaderAddr  string
 
 	networkFailure atomic.Bool
 }
 
-func newPDServiceClient(addr, leaderAddr string, conn *grpc.ClientConn, isLeader bool) ServiceClient {
+func newPDServiceClient(addr, leaderAddr string, tlsCfg *tls.Config, conn *grpc.ClientConn, isLeader bool) ServiceClient {
+	var httpAddress string
+	if strings.HasPrefix(addr, httpScheme) {
```

Comment:
What if the address starts with `http` but still has a TLS config passed in? In this case, replacing the scheme is also needed.

Reply:
Good catch. It seems that we didn't handle this case well before.